tests/integration/architecture.py

```diff
@@ -0,0 +1,7 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+import subprocess
+
+architecture = subprocess.run(
+    ["dpkg", "--print-architecture"], capture_output=True, check=True, encoding="utf-8"
+).stdout.strip()
```

Comment:
I think we should rather use `platform.machine()` and a mapping to the correct architecture. Otherwise the tests will only run on Debian derivatives without tweaking.

Reply:
Yes and it is what we use for installing the correct snap revision.